File: utils/backup/turtlebot3-backup/ros2_ws/src/py_example_package/py_example_package/dummy_subscriber.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class DummySubscriber(Node):
    def __init__(self):
        super().__init__('dummy_subscriber')

        # Declare the parameter for the topic name
        self.declare_parameter('topic_name', 'chatter')

        # Get the topic name from the parameter
        topic_name = self.get_parameter('topic_name').get_parameter_value().string_value

        # Create the subscription
        self.subscription = self.create_subscription(
            String,
            topic_name,
            self.listener_callback,
            10
        )
        self.get_logger().info(f"Subscribed to topic: {topic_name}")

    def listener_callback(self, msg):
        self.get_logger().info(f"Received message: {msg.data}")


def main(args=None):
    rclpy.init(args=args)
    node = DummySubscriber()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: dockerfiles/turtlebot3/src/c_example_package/src/dummy_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class DummySubscriber : public rclcpp::Node
{
public:
    DummySubscriber()
    : Node("dummy_subscriber")
    {
        // Declare parameter
        this->declare_parameter<std::string>("topic_name", "chatter");

        // Get the topic name from the parameter
        topic_name_ = this->get_parameter("topic_name").as_string();

        // Create subscription
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            topic_name_, 10,
            std::bind(&DummySubscriber::topic_callback, this, std::placeholders::_1));
    }

private:
    void topic_callback(const std_msgs::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "Received: '%s'", msg->data.c_str());
    }

    std::string topic_name_;
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<DummySubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
